### PR TITLE
ZK-4564: split zkau servlet into dedicated au and resource servlets

### PR DESCRIPTION
**Pull request by DevChu**

### CODE REVIEW

File: zktest/src/archive/WEB-INF/web.xml [hunk 1]
```diff
@@ -37,6 +37,10 @@ Copyright (C) 2004 Potix Corporation. All Rights Reserved.
 			<param-name>update-uri</param-name>
 			<param-value>/zkau</param-value>
 		</init-param>
+		<init-param>
+			<param-name>resource-uri</param-name>
+			<param-value>/zksr</param-value>
```
Comment (DevChu) [inline zktest/src/archive/WEB-INF/web.xml:42]: zkres

File: zk/src/org/zkoss/zk/au/http/DHtmlResourceServlet.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import org.zkoss.web.util.resource.ClassWebResource;
+import org.zkoss.zk.ui.Session;
+import org.zkoss.zk.ui.WebApp;
+import org.zkoss.zk.ui.http.I18Ns;
+import org.zkoss.zk.ui.http.SessionResolverImpl;
+import org.zkoss.zk.ui.http.WebManager;
+import org.zkoss.zk.ui.sys.SessionsCtrl;
+
+/**
+ * @author jameschu
```
Comment (DevChu) [inline zk/src/org/zkoss/zk/au/http/DHtmlResourceServlet.java:45]: since 9.1.0

File: zk/src/org/zkoss/zk/au/http/DHtmlResourceServlet.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		//Optimize the access of static resources (for GAE)
+	}
+
+	static void doGet0(HttpServletRequest request, HttpServletResponse response, ServletContext ctx,
+	                   ClassWebResource cwr) throws ServletException, IOException {
+		doGet0(request, response, ctx, cwr, false);
+	}
+
+	static void doGet0(HttpServletRequest request, HttpServletResponse response, ServletContext ctx,
+			ClassWebResource cwr, boolean fromResourceServlet) throws ServletException, IOException {
```
Comment (DevChu) [inline zk/src/org/zkoss/zk/au/http/DHtmlResourceServlet.java:119]: remove fromResourceServlet

File: zk/src/org/zkoss/zk/au/http/DHtmlUpdateServlet.java [hunk 4]
```diff
@@ -396,45 +395,10 @@ private ClassWebResource getClassWebResource() {
...
-					I18Ns.cleanup(request, old);
-				else {
-					Charsets.cleanup(request, old);
-					SessionsCtrl.setRawCurrent(oldsess);
-				}
-			}
-			return; //done
-		}
-
+		DHtmlResourceServlet.doGet0(request, response, ctx, getClassWebResource());
```
Comment (DevChu) [inline zk/src/org/zkoss/zk/au/http/DHtmlUpdateServlet.java:399]: return ?

File: zk/src/archive/web/js/zk/zk.js [hunk 5]
```diff
@@ -1280,10 +1290,10 @@ zk.log('value is", value);
 	 * @return String the encoded resource URI
 	 * @since 9.0.0
 	 */
-	resourceURI: function (uri, version) {
+	genResourceURI: function (uri, version) {
```
Comment (DevChu) [inline zk/src/archive/web/js/zk/zk.js:1293]: ajaxResourceURI:

File: zk/src/archive/web/js/zk/pkg.js [hunk 6]
```diff
@@ -158,12 +157,8 @@ zk.load('zul.utl', function () {
...
 
 		if (host) uri = host + '/web/js' + uri;
-		else {
-			if (modver) uri = '/web/_zv' + modver + '/js' + uri;
-			else uri = '/web/js' + uri;
-			uri = zk.ajaxURI(uri, {desktop: dt,au: true});
-		}
-
+		else
+			uri = zk.ajaxResourceURI('/js' + uri, zk.getVersion(pkg));
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/pkg.js:161]: `desktop: dt` was missing

File: zk/src/org/zkoss/zk/au/http/DHtmlResourceServlet.java [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+/**
+ * Used to receive ZK resource from the server
+ * <p>Init parameters:
+ * <dl>
+ * <dt>compress</dt>
+ * <dd>It specifies whether to compress the output if the browser supports the compression (Accept-Encoding).</dd>
+ * </dl>
+ * @author tomyeh
+ * @author jameschu
+ * @since 9.1.0
```
Comment (scribetw) [inline zk/src/org/zkoss/zk/au/http/DHtmlResourceServlet.java:53]: 9.2.0

File: zk/src/org/zkoss/zk/ui/impl/DesktopImpl.java [hunk 8]
```diff
@@ -232,7 +234,7 @@
 	 * If null or empty is specified, "ajax" is assumed.
 	 * @since 3.0.1
 	 */
-	public DesktopImpl(WebApp wapp, String updateURI, String path, String deviceType, Object request) {
+	public DesktopImpl(WebApp wapp, String updateURI, String resourceURI, String path, String deviceType, Object request) {
```
Comment (scribetw) [inline zk/src/org/zkoss/zk/ui/impl/DesktopImpl.java:237]: Create an overloaded constructor instead

**Pull request closed without merging**